src/main/java/com/firebolt/jdbc/statement/preparedstatement/FireboltPreparedStatement.java

```diff
@@ -198,6 +206,14 @@ public void setObject(int parameterIndex, Object x, int targetSqlType) throws SQ
 		validateStatementIsNotClosed();
 		validateParamIndex(parameterIndex);
 		try {
+			if (this.getConnection().getClass() == FireboltConnectionUserPassword.class) {
+				// We don't know the targetSqlType, so we let JavaTypeToFireboltSQLString deal
+				// with legacy conversion
+				providedParameters.put(parameterIndex,
+						JavaTypeToFireboltSQLString.transformAny(x, targetSqlType, ParserVersion.LEGACY));
+			} else {
+				providedParameters.put(parameterIndex, JavaTypeToFireboltSQLString.transformAny(x, targetSqlType));
+			}
 			providedParameters.put(parameterIndex, JavaTypeToFireboltSQLString.transformAny(x, targetSqlType));
```

Comment:
Can we have a connection class have a parser_version variable? So that we can jsut pass it in the end.
Also, it seems like we do `put` twice, once in if and second time on line 217, is that intended?

Reply:
> it seems like we do put twice
Well spotted, looks like I merged this with errors.

Reply:
As discussed offline, refactored connection class detection.